tools/last.py

```diff
@@ -35,49 +37,80 @@ class Lastal(LastTools):
     subtool_name = 'lastal'
     subtool_name_on_broad = 'lastal'
 
-
-class MafSort(LastTools):
-    """ wrapper for maf-sort subtool """
-    subtool_name = 'maf-sort'
-    subtool_name_on_broad = 'scripts/maf-sort.sh'
+    def get_hits(self, inBam, db,
+            max_gapless_alignments_per_position=1,
+            min_length_for_initial_matches=5,
+            max_length_for_initial_matches=50,
+            max_initial_matches_per_position=100
+        ):
+
+            # convert BAM to interleaved FASTQ
+            fastq_pipe = tools.samtools.SamtoolsTool().bam2fq_pipe(inBam)
+
+            # run lastal and emit list of read IDs
+            # -P 0 = use threads = core count
+            # -N 1 = report at most one alignment per query sequence
+            # -i 1G = perform work in batches of at most 1GB of query sequence at a time
+            # -f tab = write output in tab format instead of maf format
+            cmd = [self.install_and_get_path(),
+                '-n', max_gapless_alignments_per_position,
+                '-l', min_length_for_initial_matches,
+                '-L', max_length_for_initial_matches,
+                '-m', max_initial_matches_per_position,
+                '-Q', '1', '-P', '0', '-N', '1', '-i', '1G', '-f', 'tab',
+                db,
+            ]
+            cmd = [str(x) for x in cmd]
+            _log.debug('| ' + ' '.join(cmd) + ' |')
+            lastal_pipe = subprocess.Popen(cmd, stdin=fastq_pipe, stdout=subprocess.PIPE)
+
+            # strip tab output to just query read ID names and emit
+            for line in lastal_pipe.stdout:
+                line = line.decode('UTF-8').rstrip('\n\r')
+                if not line.startswith('#'):
+                    read_id = line.split('\t')[6]
+                    if read_id.endswith('/1') or read_id.endswith('/2'):
+                        read_id = read_id[:-2]
+                    yield read_id
 
 
 class Lastdb(LastTools):
     """ wrapper for lastdb subtool """
     subtool_name = 'lastdb'
     subtool_name_on_broad = 'lastdb'
 
-    def build_database(self, fasta_files, database_prefix_path): # pylint: disable=W0221
-        output_file_prefix = os.path.basename(database_prefix_path)
-        output_directory = os.path.dirname(database_prefix_path)
-
+    def is_indexed(self, db_prefix):
+        return all(os.path.exists(db_prefix + x)
+            for x in ('.bck', '.des', '.prj', '.sds', '.ssp', '.suf', '.tis'))
```

Comment:
This makes more sense here than in `taxon_filter.py`, though it could be handy to have a class property exposing the list of extensions (not that any of the other tools do this yet, but it would be a start).